File: week6/solution1.cs
//프로그래머스 
//코딩테스트 연습 - 연습문제
//하샤드 수
//https://school.programmers.co.kr/learn/courses/30/lessons/12947

public class Solution
{
    public bool solution(int x)
    {
        bool answer = false;
        string str = x.ToString();
        int parse_num = 0;
        int total = 0;

        for (int i = 0; i < str.Length; i++){
            //1. 자릿수 합 계산
            total += int.Parse(str[i].ToString());
        }

        //2. x % 자릿수 합 계산 == 0이면 하샤드 수
        if ((x % total) == 0){
            answer = true;
        }
        else{
            answer = false;
        }
        return answer;
    }
}

File: week8/solution1.cpp
//프로그래머스 
//코딩테스트 연습 - 연습문제
//최댓값과 최솟값
//https://school.programmers.co.kr/learn/courses/30/lessons/12939

#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(string s) {
    string answer = "";
    string delimit = " ";
    vector<int> token{};
    size_t pos = 0;
    int max, min;
    
    s.insert(s.length(), " ");
    
    while((pos = s.find(delimit)) != string::npos){
        token.push_back(stoi(s.substr(0, pos)));
        s.erase(0, pos + delimit.length());
    }
    
    sort(token.begin(), token.end());
    
    //for(const auto &s : token){
        //cout << s << endl;
    //}
    
    min = token.front();
    max = token.back();
    
    //cout << max << " " << min << endl;
    answer = to_string(min) + " " + to_string(max);
    
    return answer;
}